config/config.go

```diff
@@ -1166,18 +1166,19 @@ func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 // OpenstackSDConfig is the configuration for OpenStack based service discovery.
 type OpenstackSDConfig struct {
-	IdentityEndpoint string         `yaml:"identity_endpoint"`
-	Username         string         `yaml:"username"`
-	UserID           string         `yaml:"userid"`
-	Password         Secret         `yaml:"password"`
-	ProjectName      string         `yaml:"project_name"`
-	ProjectID        string         `yaml:"project_id"`
-	DomainName       string         `yaml:"domain_name"`
-	DomainID         string         `yaml:"domain_id"`
-	Role             OpenStackRole  `yaml:"role"`
-	Region           string         `yaml:"region"`
-	RefreshInterval  model.Duration `yaml:"refresh_interval,omitempty"`
-	Port             int            `yaml:"port"`
+	IdentityEndpoint   string         `yaml:"identity_endpoint"`
+	Username           string         `yaml:"username"`
+	UserID             string         `yaml:"userid"`
+	Password           Secret         `yaml:"password"`
+	ProjectName        string         `yaml:"project_name"`
+	ProjectID          string         `yaml:"project_id"`
+	DomainName         string         `yaml:"domain_name"`
+	DomainID           string         `yaml:"domain_id"`
+	Role               OpenStackRole  `yaml:"role"`
+	Region             string         `yaml:"region"`
+	RefreshInterval    model.Duration `yaml:"refresh_interval,omitempty"`
+	Port               int            `yaml:"port"`
+	AuthOptionsFromEnv bool           `yaml:"auth_options_from_env"`
```

Comment:
There should be no option by this name, as Prometheus does not officially support reading any config from anywhere other than the configuration file. What I'd suggest looking into is making this a default value for the settings like we do with AWS.

Reply:
Fair enough. I'll look into it next week.